core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -951,6 +951,25 @@ wasi_path_remove_directory(wasm_exec_env_t exec_env, wasi_fd_t fd,
     return wasmtime_ssp_path_remove_directory(curfds, fd, path, path_len);
 }
 
+static __wasi_timestamp_t
+get_timeout_for_poll_oneoff(const wasi_subscription_t *in)
+{
+    __wasi_timestamp_t timeout = 0;
+    const __wasi_subscription_t *clock_subscription = NULL;
+    if (clock_subscription == NULL
+        && (in->u.u.clock.flags & __WASI_SUBSCRIPTION_CLOCK_ABSTIME) == 0) {
+        clock_subscription = in;
+    }
+    if (clock_subscription != NULL) {
+        __wasi_timestamp_t ts = clock_subscription->u.u.clock.timeout / 1e9;
```

Comment:
So this function returns value in seconds? But then here: https://github.com/bytecodealliance/wasm-micro-runtime/pull/1951/files#diff-18a1c12794da572224835cdd9491ff7f93bc7ff81249c75ead3052e7f815f9f8R1006 we compare it to `elapsed` which is in `time_quant` unit - luckily, it's exactly a second now, but we might accidentally change the quant to be e.g. 0.5s and forget updating other places. My recommendation would be to stick to nanoseconds like we do everywhere unless we have a good reason not to do that.

Reply:
yes, that's exactly why I had kept `timestamp` as int earlier, but now it it seems we're good to get rid of the division given the datatype is changed to `__wasi_timestamp_t`